### PR TITLE
[Merged by Bors] - CI: fix mk_all

### PR DESCRIPTION
**Pull request by adomani**
This should fix the `mk_all` step in CI that leaves through PRs that do not import all files.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: .github/build.in.yml [hunk 1]
```diff
@@ -195,7 +205,12 @@ jobs:
           MATHLIB_CACHE_S3_TOKEN: ${{ secrets.MATHLIB_CACHE_S3_TOKEN }}
 
       - name: update {Mathlib, Tactic, Counterexamples, Archive}.lean
-        run: lake exe mk_all --check
+        run: |
+          if [ ${{ steps.mk_all.outputs.mk_all }} == "false" ]
+          then
+            echo "Please run 'lake exe mk_all' to regenerated the import all files"
```
Comment (Ruben-VandeVelde) [inline .github/build.in.yml:211]:
```suggestion
            echo "Please run 'lake exe mk_all' to regenerate the import all files"
```

**Review by Ruben-VandeVelde — APPROVED** [review]

**Comment by adomani** [comment]
Ruben, thanks!  I'm about to board a plane and will not have time for further tests.  If you feel like it, please jump in and fix any mistakes that you may find!

**Comment by bryangingechen** [comment]
This seems like a reasonable approach to take. I guess the previous version wasn't working because after the first `mk_all` was run, the second one was just checking that output!

When you're back please run any other tests you were thinking of and then merge. Thanks!
bors d+

(I was puzzled as to why the "lint and suggest" workflow was not suggesting any fixes to your test ([log](https://github.com/leanprover-community/mathlib4/actions/runs/10908363736/job/30274156961?pr=16896)) and then I remembered that GitHub [does not allow](https://github.com/orgs/community/discussions/9099) PR comments on lines that are not touched by the PR!)

**Comment by semorrison** [comment]
bors merge

**Comment by bryangingechen** [comment]
bors r-

@semorrison Sorry, this still has a test commit on it!

**Comment by adomani** [comment]
I wanted to do one more test: verify that with modifications only in already existing files, the new steps did not flag anything.  This was successful.

I'll merge as soon as CI is green again!

**Pull request closed without merging**